### DISCUSSION TITLE
[x] [Topic 2 Day 2] Career development as an RSE

### DISCUSSION THREAD

**Discussion by rowlandm** [RSEAA/RSEAA.github.io#39] [Discussion topics for #RSEAA2023]
This is always something that gets brought up. But what does "Having a career as an RSE" actually mean for RSEs? What are the concrete things that would make you say "Yes, there is a career as an RSE".

**Comment by betsybookwyrm** [comment]
following on from yesterday - what does "top level" look like? but that would look different at a research institution than at a commercial institution that just happens to do research

**Reply by aportagain** [reply]
I think it's definitely interesting and useful to consider different potential top steps in the career ladder / progression, like in the related discussion yesterday people doing RSE who reached professorship (https://github.com/RSEAA/RSEAA.github.io/discussions/44#discussioncomment-6985536). Even if just because from concrete examples like that, you can trace back potential pathways. In less academic environments, I've definitely seen "Scientific Software Architect" pop up once or twice as a job title in recent years. Would love to hear if there's any examples of a "Principal Something" RSE role.

**Comment by betsybookwyrm** [comment]
separation or not of people management from career progression?

**Comment by betsybookwyrm** [comment]
HEW levels (australian professional university staff tiers) don't really correspond to professional development, but then associate dev/dev/senior/principal etc in industry isn't necessarily meaningful either

**Comment by betsybookwyrm** [comment]
in Africa RSE is not really a known term - there is an RSE group in South Africa, but Uganda and DR Congo and other places it's not known

**Reply by betsybookwyrm** [reply]
The term isn't widely known in Australia and NZ either - but the concept exists, so you can just explain that the term refers to the known concept

**Comment by betsybookwyrm** [comment]
what does RSE even mean? research software development is unique and has its own skills, common themes, etc that is different from other software development

**Reply by betsybookwyrm** [reply]
RSE is a type of software development where you don't want to delegate the domain knowledge to a separate person

**Reply by betsybookwyrm** [reply]
A lot of codesign where the developers aren't domain experts

**Reply by betsybookwyrm** [reply]
people from the tech space moving into the RSE space need to learn about working with domain experts and how the academic context differs and has different pressures
people from the research space moving into RSE need to learn skills for writing software for/with others beyond skills they used in their own research code

**Reply by nrmay** [reply]
@betsybookwyrm re. RSE: the software engineering processes are the same in the research domain as any other, such as banking, logistics, telecoms, finance, etc. Software engineers have to work with the subject matter experts (SMEs) to elicit and refine the software requirements without having to learn the subject in depth.

What makes RSE different is the variety of subjects it has to cover and the high level of novice practitioners (often the SMEs themselves) that are performing the software engineering roles. The former means that requirement engineering skills are more important. The latter means that a substantial amount of the design/development work is performed by people with little or no software engineering education and expertise. Often, when the RSE community identifies a problem that it wants to address (abstraction, interoperability, etc.), there is already a substantial, well documented body of knowledge that applies to that problem, such as: https://en.wikipedia.org/wiki/Software_Engineering_Body_of_Knowledge

**Comment by aportagain** [comment]
In today's review session, @manodeep mentioned the example of King's Digital Lab who published their research software careers structure / progression a few years ago (2019?):
https://kdl.kcl.ac.uk/blog/rse-career-development/
http://doi.org/10.5281/zenodo.2559235
